tests/test_basic_ops_dict.py

```diff
@@ -84,6 +84,12 @@ def test_subscript_get() -> None:
     assert "b" == c["a"]
 
 
+def test_subscript_get_int_key() -> None:
+    c = OmegaConf.create("1: b")
+    assert isinstance(c, DictConfig)
+    assert "b" == c[1]
```

Comment:
Create from dictionary instead of from string.
```suggestion
def test_subscript_get_int_key() -> None:
    c = OmegaConf.create({1: b})
    assert isinstance(c, DictConfig)
    assert "b" == c[1]
```